lib/generators/rolemodel/saas/blazer/blazer_generator.rb

```diff
@@ -0,0 +1,71 @@
+require_relative '../../../bundler_helpers'
+
+module Rolemodel
+  module Saas
+    class BlazerGenerator < Rails::Generators::Base
+      include Rolemodel::BundlerHelpers
+      source_root File.expand_path('templates', __dir__)
+
+      def install_blazer
+        gem 'sprockets-rails' unless File.readlines('Gemfile').grep(/sprockets/).any?
+        gem 'blazer'
+        run_bundle
+
+        generate 'blazer:install'
+      end
+
+      def update_migration
+        filename = Dir.glob('db/migrate/*_install_blazer.rb').first
+        inject_into_file filename, ', foreign_key: { to_table: :users }', after: 't.references :creator'
+      end
+
+      def add_routes
+        return if File.readlines('config/routes.rb').grep(/blazer/).any?
+
+        # add routes for the report controllers
+        route_info = "  namespace :reports do\n"
+        route_info += "    resources :dashboards, only: %i[index show]\n"
+        route_info += "    resources :queries, only: [] do\n"
+        route_info += "      post :run, on: :collection\n"
+        route_info += "      # post :cancel, on: :collection\n"
+        route_info += "    end\n"
+        route_info += "  end\n"
+        route route_info
+
+        # add routes for the Blazer engine
+        route_info = "  # authenticate :user, ->(u) { Admin::ReportPolicy.new(u, :report).manage? } do\n"
+        route_info += "  mount Blazer::Engine => '/admin/reports', as: :blazer\n"
+        route_info += "  # end\n"
+        route route_info
+      end
+
+      def add_extensions
+        copy_file 'config/initializers/blazer.rb'
+        copy_file 'lib/blazer_extensions/data_source.rb'
+      end
+
+      def add_controllers
+        directory 'app/controllers/reports'
+      end
+
+      def add_views
+        directory 'app/views/reports'
+      end
+
+      def add_styles
+        copy_file 'app/assets/stylesheets/blazer.css'
+        copy_file 'app/assets/stylesheets/selectize.css'
+        append_to_file 'app/assets/config/manifest.js', "\n//= link blazer.css"
```

Comment:
This will be different base on the CSS approach being used right?

Reply:
I defer to @reedrolemodel on that one. I forget why we needed that.

Reply:
Yes, this assumes `sprockets` is being used which is also an assumption of `blazer`. [This issue](https://github.com/ankane/blazer/issues/379) is for adding support for `jsbundling` and `cssbundling` to `blazer`. Until more gems support newer approaches, it seems likely that `sprockets` will be needed.